helidon-maven-plugin/src/main/java/io/helidon/build/maven/graal/GraalNativeMojo.java

```diff
@@ -279,6 +316,69 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         }
     }
 
+    /**
+     * Build module-path and class-path. Both key will be present in the returned map.
+     *
+     * @return map containing class-path and module-path
+     */
+    private Map<String, String> buildModuleAndClassPath() {
+        getLog().debug("Building module-path string");
+        List<String> modules = new LinkedList<>();
+        List<String> cp = new LinkedList<>();
+        File jarFile = new File(buildDirectory, finalName + ".jar");
+        LocationManager locationManager = new LocationManager();
+
+        Optional<SourcePath> moduleDescriptor = SourcePath.scan(Path.of(project.getBuild().getSourceDirectory()).toFile())
+                .stream()
+                .filter(p -> p.matches("module-info.java"))
+                .findAny();
+
+        if (jarFile.exists()) {
+            if (moduleDescriptor.isPresent()) {
+                modules.add(jarFile.getAbsolutePath());
+            } else {
+                cp.add(jarFile.getAbsolutePath());
+            }
+        } else {
+            getLog().debug(String.format("Jar file %s does not exist, won't be present on module/class path", jarFile.getName()));
```

Comment:
Ins't this a warning, or maybe info ? This would be happening if the user invokes the plugin in the wrong phase.